res/res_rtp_asterisk.c

```diff
@@ -9135,26 +9141,31 @@ static void ast_rtp_remote_address_set(struct ast_rtp_instance *instance, struct
  */
 static int red_write(const void *data)
 {
-	struct ast_rtp_instance *instance = (struct ast_rtp_instance*) data;
-	struct ast_rtp *rtp = ast_rtp_instance_get_data(instance);
+	struct ast_rtp_instance *instance = (struct ast_rtp_instance *) data;
+	struct ast_rtp *rtp;
 
 	ao2_lock(instance);
-	if (rtp->red->t140.datalen > 0) {
-		ast_rtp_write(instance, &rtp->red->t140);
+	rtp = ast_rtp_instance_get_data(instance);
+	if (!rtp || !rtp->red) {
+		ao2_unlock(instance);
+		ast_log(LOG_ERROR, "Could not get rtp from instance or RED handler\n");
+		return 0;
 	}
+
 	ao2_unlock(instance);
 
 	return 1;
 }
 
 /*! \pre instance is locked */
-static int rtp_red_init(struct ast_rtp_instance *instance, int buffer_time, int *payloads, int generations)
+static int rtp_red_init(struct ast_rtp_instance *instance, int buffer_time, int payload, int generations)
 {
 	struct ast_rtp *rtp = ast_rtp_instance_get_data(instance);
 	int x;
 
 	rtp->red = ast_calloc(1, sizeof(*rtp->red));
 	if (!rtp->red) {
+		ast_log(LOG_ERROR, "Could not get RED handler\n");
```

Comment:
Allocation errors are already logged so nothing needs to be output here.

Reply:
Sure, will be removed